prov/verbs/src/ep_rdm/verbs_ep_rdm.c

```diff
@@ -435,6 +441,7 @@ static int fi_ibv_rdm_ep_close(fid_t fid)
 		av_entry = container_of(item,
 					struct fi_ibv_rdm_av_entry,
 					removed_next);
+		pthread_mutex_lock(&av_entry->conn_lock);
 		HASH_FIND(hh, av_entry->conn_hash, &ep,
 			  sizeof(struct fi_ibv_rdm_ep *), conn);
 		if (conn) {
```

Comment:
Same here regarding connection cleanup.

Reply:
Sure, we can do this by adding ugly if-else statement. But it should be okay